azurerm/resource_arm_netapp_volume.go

```diff
@@ -0,0 +1,357 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/netapp/mgmt/2019-06-01/netapp"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	aznetapp "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/netapp"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmNetAppVolume() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmNetAppVolumeCreateUpdate,
+		Read:   resourceArmNetAppVolumeRead,
+		Update: resourceArmNetAppVolumeCreateUpdate,
+		Delete: resourceArmNetAppVolumeDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: aznetapp.ValidateNetAppVolumeName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: aznetapp.ValidateNetAppAccountName,
+			},
+
+			"pool_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: aznetapp.ValidateNetAppPoolName,
+			},
+
+			"creation_token": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: aznetapp.ValidateNetAppVolumeCreationToken,
+			},
+
+			"service_level": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(netapp.Premium),
+					string(netapp.Standard),
+					string(netapp.Ultra),
+				}, true),
+			},
+
+			"subnet_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"usage_threshold": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(100, 4096),
+			},
+
+			"export_policy_rule": {
+				Type:     schema.TypeList,
```

Comment:
since the rule_index is part of this presumably these could be returned in any order and `export_policy_rule` needs to become a Set?

Reply:
I found *_rule also is set as TypeList in existing resources of terraform. Why i cannot use TypeList for this field like others? For more details about rule_index and export_policy, please see this doc https://docs.microsoft.com/en-us/azure/azure-netapp-files/azure-netapp-files-configure-export-policy

Reply:
Most of those should probably be `TypeSets` too, as order doesn't matter if `rule_index` is used, otherwise if we keep it as a typelist the ordering could be implicitly inferred by the index. WDYT @tombuildsstuff ?

Reply:
fixed